src/statusicon.h

```diff
@@ -16,37 +16,32 @@ class StatusIcon final : public QObject {
   Q_OBJECT
   Q_DISABLE_COPY_MOVE(StatusIcon)
 
-  Q_PROPERTY(QUrl iconUrl READ iconUrl NOTIFY iconChanged)
-
  public:
   StatusIcon();
   ~StatusIcon();
 
-  QUrl iconUrl() const;
-  const QString& iconString() const { return m_iconUrl; }
   const QIcon& icon();
-  const QColor& indicatorColor() const { return m_indicatorColor; }
+  const QString& iconString() const { return m_iconUrl; }
+  const QColor& indicatorColor();
 
  signals:
-  void iconChanged();
-  void indicatorColorChanged();
+  void iconUpdateNeeded();
 
  public slots:
   void stateChanged();
-  void stabilityChanged();
 
  private slots:
   void animateIcon();
 
  private:
   void activateAnimation();
   QIcon drawStatusIndicator(const QString& iconUrl) const;
-  void setIcon(const QString& iconUrl, bool shouldDrawIndicator);
-  void setIndicatorColor(const QColor& color);
+  void generateIcon();
+  void setIcon(const QString& iconUrl);
 
  private:
-  QString m_iconUrl;
   QIcon m_icon;
+  QString m_iconUrl;
```

Comment:
You can remove this or to optimize the code, keep it and use it as a cache like m_icon.
In stateChanged, set it to null.